pkg/pillar/cmd/downloader/resolveconfig.go

```diff
@@ -275,7 +276,11 @@ func resolveTagsToHash(ctx *downloaderContext, rc types.ResolveConfig,
 				errStr = "tag resolution cancelled by user"
 				break
 			}
-			errStr = errStr + "\n" + err.Error()
+			if strings.Contains(err.Error(), logutils.NoSuitableAddrStr) {
```

Comment:
This is not the go way to look at errors; an error isn't a string just because it has a string rendition. Folks have written about better ways. See e.g., https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully

I don't know the best way to do this using the old net package, but a slightly better is to traverse the error hiearchy. When looking for an example I found isNoSuitableAddress() in send.go.
Note that it still ends at a string compare because the standard net package does not export the variable (or constant) which has that string rendition.

Reply:
actually i used the isNoSuitableAddress() for http scheme, but it breaks in the e2 which trying to check the AddrError and to be not ok. I tried to find if there is any other error scheme under e1.Err, but could not find. and for azure, aws, docker, those errors don't even fit into url.Error(), but I didn't try to explore all the error available

Reply:
Seems like too many go packages don't follow the golang recommendations on how to generate and handle errors, Sigh,